lib/rules/no-restricted-v-on.js

```diff
@@ -0,0 +1,192 @@
+'use strict'
+
+const utils = require('../utils')
+const regexp = require('../utils/regexp')
+
+const DEFAULT_OPTIONS = [
+  {
+    argument: '/^v-/',
+    message:
+      'Using `@v-xxx` is not allowed. Instead, remove `@` and use it as directive.'
+  }
+]
+/**
+ * @typedef {object} ParsedOption
+ * @property { (key: VDirectiveKey) => boolean } test
+ * @property {string[]} modifiers
+ * @property {boolean} [useElement]
+ * @property {string} [message]
+ */
+
+/**
+ * @param {string} str
+ * @returns {(str: string) => boolean}
+ */
+function buildMatcher(str) {
+  if (regexp.isRegExp(str)) {
+    const re = regexp.toRegExp(str)
+    return (s) => {
+      re.lastIndex = 0
+      return re.test(s)
+    }
+  }
+  return (s) => s === str
+}
+
+/**
+ * @param {any} option
+ * @returns {ParsedOption}
+ */
+function parseOption(option) {
+  if (typeof option === 'string') {
+    const matcher = buildMatcher(option)
+    return {
+      test(key) {
+        return Boolean(
+          key.argument &&
+            key.argument.type === 'VIdentifier' &&
+            matcher(key.argument.rawName)
+        )
+      },
+      modifiers: []
+    }
+  }
+  if (option === null) {
+    return {
+      test(key) {
+        return key.argument === null
+      },
+      modifiers: []
+    }
+  }
+  const parsed = parseOption(option.argument)
+
+  if (option.modifiers) {
+    const argTest = parsed.test
+    parsed.test = (key) => {
+      if (!argTest(key)) {
+        return false
+      }
+      return /** @type {string[]} */ (option.modifiers).every((modName) =>
+        key.modifiers.some((mid) => mid.name === modName)
+      )
+    }
+    parsed.modifiers = option.modifiers
+  }
+  if (option.element) {
+    const argTest = parsed.test
+    const tagMatcher = buildMatcher(option.element)
+    parsed.test = (key) => {
+      if (!argTest(key)) {
+        return false
+      }
+      return tagMatcher(key.parent.parent.parent.rawName)
+    }
+    parsed.useElement = true
+  }
+  parsed.message = option.message
+  return parsed
+}
+
+/**
+ * @param {VDirectiveKey} key
+ * @param {ParsedOption} option
+ */
+function defaultMessage(key, option) {
+  const von = key.name.rawName === '@' ? '' : 'v-on'
+  const arg =
+    key.argument != null && key.argument.type === 'VIdentifier'
+      ? `@${key.argument.rawName}`
+      : ''
+  const mod =
+    option.modifiers.length > 0 ? `.${option.modifiers.join('.')}` : ''
+  let element = 'element'
+  if (option.useElement) {
+    element = `<${key.parent.parent.parent.rawName}>`
+  }
+  return `Using \`${von + arg + mod}\` is not allowed on this ${element}.`
+}
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow specific argument in `v-on`',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-restricted-v-on.html'
+    },
+    fixable: null,
+    schema: {
+      type: 'array',
+      items: {
+        oneOf: [
+          { type: ['string', 'null'] },
```

Comment:
Why is `null` allowed?

Reply:
When you added null, it would use the `DEFAULT_OPTIONS` options  as an additional argument, I've removed this as it's no longer necessary

Reply:
Ah, I remember we also allow `null` for the `no-restricted-v-bind` rule to disallow `v-bind` without an argument. So `null` should indeed be a possible option here that can be set to disallow `v-on` without an argument (e.g. `v-on="listeners"`). Sorry for not checking you test cases earlier.

Reply:
Cool no problem, I've addressed this @FloEdelmann  🦦:
- Updated docs
- Added another test case for this condition